pkg/daemon/on_disk_validation.go

```diff
@@ -47,9 +154,39 @@ func validateOnDiskState(currentConfig *mcfgv1.MachineConfig, systemdPath string
 	}
 }
 
+func checkV3Passwd(sshKeys []ign3types.SSHAuthorizedKey, filePath string) error {
```

Comment:
suggestion: Instead of passing just the SSH keys to this function, why not pass the whole `Passwd` object?

Reply:
AFAIK Currently the MC config does not support adding users or their SSH keys other than `Core`. Hence after passing the `Passwd` object I would still be checking only for the first core user?